### PR TITLE
Inital updates for GNOME 40

### PR DESCRIPTION
**Pull request by Oreo639**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64-glibc)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

Obviously, this depends on https://github.com/void-linux/void-packages/pull/29689

### CODE REVIEW

**Comment by ericonr** [comment]
Feel free to allow CI to run ;)

Though we should probably merge glib soon so the results make any sense :p I will take a look at their current bug reports to see if waiting for 2.68.1 is actually necessary.

**Comment by Oreo639** [comment]
CI will fail due to a too low version of glib being available. And yes, it is needed. (iirc it is required for mutter)

**Review by paper42 — COMMENTED** [review]
`vsed -i "s/pam_systemd\.so/pam_elogind.so/" $DESTDIR/etc/pam.d/*` in gdm doesn't do anything

File: srcpkgs/gtk4/template [hunk 1]
```diff
@@ -29,7 +29,7 @@ maintainer="Enno Boland <gottox@voidlinux.org>"
 license="LGPL-2.1-or-later"
 homepage="https://www.gtk.org/"
 distfiles="${GNOME_SITE}/gtk/${version%.*}/gtk-${version}.tar.xz"
-checksum=d46cf5b127ea27dd9e5d2ff6ed500cb4067eeb2cb1cd2c313ccde8013b0b9bf9
+checksum=e975f286e911666a79b6bcf486e6f99b0bd9d2b4cc348d19bce487a0b1c97072
 
 # Package build options
 build_options="broadway colord cups gir cloudproviders wayland x11"
```
Comment (paper42) [inline srcpkgs/gtk4/template:35]: "vulkan" option is missing

**Comment by paper42** [comment]
I have been running GNOME 40 on my computer for a week (updated templates are in my fork's master branch), but there have been some problems:
* the shell crashed 2 times, I don't have  a backtrace yet
* every workspace preview on top of the screen shows preview of the same workspace from my secondary monitor
* I had to add a small sleep to the gdm service on one of my computers, because elogind was not starting fast enough. sv check elogind in the service should fix this.
* gnome-control-center should have libhandy1-devel in makedepends

**Comment by st3r4g** [comment]
> sv check elogind in the service should fix this

But that requires a check script for elogind, right? Try the following check script `/etc/sv/elogind/check`:
```
#!/bin/sh
exec dbus-send --print-reply --reply-timeout=1000 --system --dest=org.freedesktop.login1 / org.freedesktop.DBus.Peer.Ping >/dev/null 2>&1
```

**Comment by ericonr** [comment]
The issue with a check script is that docs say dbus activation is an option for elogind :/

I'm not sure how we should solve this.

@paper42 are you on musl or glibc? There might be some useful patches on Alpine.

**Comment by paper42** [comment]
> The issue with a check script is that docs say dbus activation is an option for elogind :/
> 
> I'm not sure how we should solve this.
> 
> @paper42 are you on musl or glibc? There might be some useful patches on Alpine.

I have GNOME 40 on 2 computers, one of them is an old thinkpad x220 and the second one is a new Ryzen 7 laptop. Both computers run glibc versions of Void, but only my x220 has this problem. A few months ago I noticed the same thing sometimes happening with gdm 3, so this is not a new bug. Before the elogind service managed to start, elogind was started by gdm, so the runit service kept failing. For some reason even though elogind was running, gdm crashed and I think I remember the logs complaining about elogind, but I am not 100% sure.

File: srcpkgs/gtk4/template (hunk 1 above)
Comment (Oreo639) [reply srcpkgs/gtk4/template:35]: Thank you.

**Comment by Oreo639** [comment]
Superseded by https://github.com/void-linux/void-packages/pull/29974

**Pull request closed without merging**